docs/cache/pattern/api-caching.md

```diff
@@ -0,0 +1,193 @@
+For code reusability, we use [the Decorator pattern](https://en.wikipedia.org/wiki/Decorator_pattern) to wrap `CachingWeatherDescriptionClient` around an abstract `WeatherDescriptionClient` that we have created in an earlier step. Note that this requires no code-change to the existing `WolframWeatherDescriptionClient`.
+```typescript
+import {CacheClient, CacheGet} from '@gomomento/sdk';
+
+// This is the name of the distributed cache that we will use
+// to store weather results for different cities.
+// This cache must be created beforehand.
+const CACHE_NAME = 'weather-descriptions';
+
+// Weather data is highly cacheable since it is not likely to change rapidly.
+// Here we can configure the TTL for how long we want to cache the weather data for.
+const WEATHER_CACHE_TTL_SECONDS = 300;
+
+// This is an inline distributed cache.
+// Results are served from the cache whenever possible.
+// On a cache miss, the actual service is called, then
+// the result will be populated into the cache automatically.
+export class CachingWeatherDescriptionClient
+  implements WeatherDescriptionClient
+{
+  delegate: WeatherDescriptionClient;
+  momentoCacheClient: IMomentoCache;
+
+  constructor(
+    delegate: WeatherDescriptionClient,
+    momentoCacheClient: CacheClient
+  ) {
+    this.delegate = delegate;
+    this.momentoCacheClient = momentoCacheClient.cache(CACHE_NAME);
+  }
+
+  // With our Momento Cache, we need to establish a keyspace to make sure
+  // that two different cities do not map to the same key.
+  // In this demo we simply use the string name of the city.
+  cacheKey(city: City): string {
+    const cityName = City[city];
+    return cityName;
+  }
+
+  // Asynchronously stores the result into cache to not impact
+  // the critical path of the application.
+  // For more comprehensive error handling,
+  // you may add .then() method to Promise<CacheSet.Response>.
+  populateCacheWithResult(city: City, result: string) {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.momentoCacheClient.set(this.cacheKey(city), result, {
+      ttl: WEATHER_CACHE_TTL_SECONDS,
+    });
+  }
+
+  async getWeatherDescription(city: City): Promise<string> {
+    const getResponse = await this.momentoCacheClient.get(this.cacheKey(city));
+
+    const weatherDescription = getResponse.value();
+    if (!weatherDescription) {
+      // If the weather is not in the cache, call actual client,
+      // and populate the cache.
+      const response = await this.delegate.getWeatherDescription(city);
+      this.populateCacheWithResult(city, response);
+      return response;
+    }
+
+    return weatherDescription;
+  }
+}
+```
+5. By using the Decorator Pattern, we can easily compose `CachingWeatherDescriptionClient` and `WolframWeatherDescriptionClient` together to create a new client that does read-aside caching, i.e. if a result is the cache, return the result, otherwise fallback to retrieving the result from the remote API.
```

Comment:
I missed this the first time, you're missing a word "if a result is **in** the cache"

Reply:
ooof, good catch, ty